core/src/replay_stage.rs

```diff
@@ -2368,9 +2368,19 @@ impl ReplayStage {
                 last_voted_slot
             );
         }
+
+        let last_vote_tx_blockhash =
+            if let Some(last_vote_tx_blockhash) = tower.last_vote_tx_blockhash() {
+                last_vote_tx_blockhash
+            } else {
+                // If we are a non voting validator or have an incorrect setup preventing us from
+                // generating vote txs, no need to refresh
+                return;
+            };
```

Comment:
The somewhat new `let-else` [feature](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) is pretty slick for situations like this
```suggestion
        // If we are a non-voting node or have an invalid setup preventing us from
        // generating valid vote transactions, no need to refresh votes
        let Some(last_vote_tx_blockhash) = tower.last_vote_tx_blockhash() else { return };
```

Reply:
cool stuff!